src/cmd-sign

```diff
@@ -0,0 +1,305 @@
+from fedora_messaging.config import conf
+
+from twisted.internet import reactor
+
+sys.path.insert(0, os.path.dirname(os.path.abspath(__file__)))
+from cosalib.meta import GenericBuildMeta as Meta
+from cosalib.builds import Builds
+from cosalib.cmdlib import (
+    get_basearch,
+    sha256sum_file,
+    import_ostree_commit)
+
+gi.require_version('OSTree', '1.0')
+from gi.repository import Gio, OSTree
+
+FEDORA_MESSAGING_TOPIC_PREFIX = "org.fedoraproject.prod.coreos.build.request"
+ROBOSIGNATORY_REQUEST_TIMEOUT_SEC = 60 * 10
+
+def main():
+    args = parse_args()
+    args.func(args)
+
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--build", help="Build ID", default='latest')
+    subparsers = parser.add_subparsers(dest='cmd', title='subcommands')
+    subparsers.required = True
+
+    robosig = subparsers.add_parser('robosignatory', help='sign with '
+                                    'RoboSignatory via fedora-messaging')
+    robosig.add_argument("--s3", metavar='<BUCKET>[/PREFIX]', required=True,
+                         help="Bucket and prefix to S3 builds/ dir")
+    group = robosig.add_mutually_exclusive_group(required=True)
+    group.add_argument("--ostree", help="Sign commit", action='store_true')
+    group.add_argument("--images", help="Sign images", action='store_true')
+    robosig.add_argument("--extra-fedmsg-keys", action='append',
+                         metavar='KEY=VAL', default=[])
+    robosig.add_argument("--fedmsg-conf", metavar='CONFIG.TOML')
+    robosig.add_argument("--gpgkeypath", help="Path to directory containing "
+                         "public keys to use for signature verification",
+                         default="/etc/pki/rpm-gpg")
+    robosig.set_defaults(func=cmd_robosignatory)
+
+    return parser.parse_args()
+
+
+def cmd_robosignatory(args):
+    builds = Builds()
+    if args.build == 'latest':
+        args.build = builds.get_latest()
+
+    s3 = boto3.client('s3')
+    args.bucket, args.prefix = get_bucket_and_prefix(args.s3)
+
+    args.extra_keys = {}
+    for keyval in args.extra_fedmsg_keys:
+        key, val = keyval.split('=', 1)  # will throw exception if there's no =
+        args.extra_keys[key] = val
+
+    if args.fedmsg_conf:
+        conf.load_config(args.fedmsg_conf)
+
+    global request_state
+    request_state = {"status": "pending"}
+    cond = threading.Condition()
+    start_consumer_thread(cond, {
+        'build_id': args.build,
+        'basearch': get_basearch(),
+        **args.extra_keys
+    })
+
+    # these two are different enough that they deserve separate handlers
+    if args.ostree:
+        robosign_ostree(args, s3, cond)
+    else:
+        assert args.images
+        robosign_images(args, s3, cond)
+
+
+def robosign_ostree(args, s3, cond):
+    build = Meta(build=args.build)
+    builds = Builds()
+    builddir = builds.get_build_dir(args.build)
+    checksum = build['ostree-commit']
```

Comment:
Seems like we could just use the Build() library for this:

```
build = Build(args.build)
builddir = build.build_dir
checksum = build.meta['ostree-commit']
...
...
build.meta_write()

Reply:
I was using the example from the cmd-buildextend-azure, but Build() is a class defined just within that file so this isn't relevant.